pkg/processor/build/runtime/java/runtime.go

```diff
@@ -0,0 +1,364 @@
+		}
+	}
+
+	if len(filesToCopy) == 0 {
+		return errors.Errorf("Can't find Java files in %q", buildPath)
+	}
+
+	for _, filePath := range filesToCopy {
+		destPath := path.Join(javaSrcDirPath, path.Base(filePath))
+		if err := util.CopyFile(filePath, destPath); err != nil {
+			return errors.Wrap(err, "Can't copy file to Java source directory")
+		}
+	}
+
+	return nil
+}
+
+func (j *java) copyJavaSources(buildPath, stagingBuildDir string) error {
+	switch {
+	case common.IsDir(buildPath):
+		if _, err := util.CopyDir(buildPath, stagingBuildDir); err != nil {
+			return errors.Wrapf(err, "Can't copy sources %q -> %q", buildPath, stagingBuildDir)
+		}
+	case j.isFile(buildPath, ".java"):
+		javaSrcDir := path.Join(stagingBuildDir, "src/main/java")
+		if err := os.MkdirAll(javaSrcDir, 0777); err != nil {
+			return err
+		}
+		destSrcPath := path.Join(javaSrcDir, path.Base(buildPath))
+		if err := util.CopyFile(buildPath, destSrcPath); err != nil {
+			return errors.Wrapf(err, "Can't copy source %q -> %q", buildPath, destSrcPath)
+		}
+	default:
+		return errors.Errorf("Don't know how to build %q", buildPath)
+	}
+
+	return nil
+}
+
+func (j *java) findHandlerJar(dirName string) (string, error) {
+	var jarFiles []string
+	walkFunc := func(path string, info os.FileInfo, err error) error {
+		if j.isFile(path, ".jar") && !j.isSDKJar(path) {
+			jarFiles = append(jarFiles, path)
+		}
+		return nil
+	}
+
+	if err := filepath.Walk(dirName, walkFunc); err != nil {
+		return "", err
+	}
+
+	switch len(jarFiles) {
+	case 1:
+		return jarFiles[0], nil
+	case 0:
+		return "", nil
+	default:
+		return "", errors.Errorf("too many jar files: %v", jarFiles)
+	}
+}
+
+func (j *java) isSDKJar(jarPath string) bool {
+	return j.isFile(jarPath, ".jar") && strings.HasPrefix(path.Base(jarPath), "nuclio-sdk-")
+}
+
+func (j *java) buildUserJar(buildPath, userJarPath string) error {
+	// If it's a jar - use it
+	if j.isFile(buildPath, ".jar") {
+		j.Logger.InfoWith("Using existing jar", "path", buildPath)
+		return util.CopyFile(buildPath, userJarPath)
+	}
+
+	// If we have handler.jar in this directory - use it
+	jarFilePath := path.Join(buildPath, "handler.jar")
+	if common.IsFile(jarFilePath) {
+		j.Logger.InfoWith("Using existing jar", "path", jarFilePath)
+		return util.CopyFile(jarFilePath, userJarPath)
+	}
+
+	var err error
+
+	// If it's a directory with a single jar file - use it (case of archives)
+	jarFilePath, err = j.findHandlerJar(buildPath)
+	if jarFilePath != "" && err == nil {
+		j.Logger.InfoWith("Using existing jar", "path", jarFilePath)
+		return util.CopyFile(jarFilePath, userJarPath)
+	}
+
+	return j.buildUserJarFromSource(buildPath, userJarPath)
+}
+
+func (j *java) buildUserJarFromSource(buildPath, userJarPath string) error {
+	userBuildDir := path.Join(j.StagingDir, "java-user-build")
+
+	if err := j.copyJavaSources(buildPath, userBuildDir); err != nil {
+		return err
+	}
+
+	err := j.createBuildFile(userBuildDir)
```

Comment:
I think it's quite important for debugging to always log the contents of this file as debug

Reply:
Done at `createBuildFile`